gobblin-compaction/src/main/java/gobblin/compaction/conditions/RecompactionConditionBasedOnTime.java

```diff
@@ -0,0 +1,33 @@
+package gobblin.compaction.conditions;
+
+
+import gobblin.compaction.dataset.DatasetMetric;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+/**
+ * An implementation {@link RecompactionCondition} which checks the earliest file modification timestamp from
+ * the late output directory. If the earliest file has been existed for a while and never been cleaned up, a
+ * recmpaction will be triggered based on how long the file has been there.
+ */
+public class RecompactionConditionBasedOnTime implements RecompactionCondition {
```

Comment:
Rename it to RecompactionConditionBasedOnDuration. It helps calculate how long a late file existed in late output directory, and based on duration, if a recompaction is necessary.